python/hidet/graph/ops/definitions/activation.py

```diff
@@ -87,6 +88,131 @@ def __init__(self, x: Tensor):
         )
 
 
+class ThresholdOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, threshold_val: float, value: float) -> Tensor:
+        super().__init__(x, op=lambda v: if_then_else(v > x.dtype(threshold_val), v, x.dtype(value)), name='threshold')
+
+
+class HardTanhOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, min_val: float = -1.0, max_val: float = 1.0) -> Tensor:
+        super().__init__(
+            x,
+            op=lambda v: if_then_else(
+                v > x.dtype(max_val), x.dtype(max_val), if_then_else(v < x.dtype(min_val), x.dtype(min_val), v)
```

Comment:
consider to use the min/max primitive to implement this:
``out = max(min(v, min_val), max_val)`,
as it will only read v once.

Reply:
refactored to:

min(max_val, max(minval, v))